src/InstrumentationEngine.Api/InstrumentationEngine.h

```diff
@@ -1972,6 +1980,263 @@ EXTERN_C const IID IID_IInstruction;
 #endif 	/* __IInstruction_INTERFACE_DEFINED__ */
 
 
+#ifndef __IInstruction2_INTERFACE_DEFINED__
+#define __IInstruction2_INTERFACE_DEFINED__
+
+/* interface IInstruction2 */
+/* [unique][uuid][object] */ 
+
+
+EXTERN_C const IID IID_IInstruction2;
+
+#if defined(__cplusplus) && !defined(CINTERFACE)
+    
+    MIDL_INTERFACE("A77D20D8-B261-41C7-9B0A-F5FBBAABA576")
+    IInstruction2 : public IInstruction
+    {
+    public:
+        virtual IInstruction2 *STDMETHODCALLTYPE GetNextInstruction( void) = 0;
```

Comment:
Per our offline discussion, we should consider a "return array of instructions type" api

Reply:
I was going to try to test the perf improvements of this API, but it turns out that the our unit tests don't call GetNextInstruction. I'll try to mock something up.

Reply:
I tested this out, at it turns out that even for large functions, this doesn't make that much of a difference in terms of time taken to iterate. If we want to make it faster, we will probably need an iterator class that returns arrays of instructions. I'll get rid of this API.